scripts/load-gotland.sh

```diff
@@ -0,0 +1,170 @@
+
+	------------------------------------------------------------
+	-- Insert
+	------------------------------------------------------------
+
+	-- Genebank
+	INSERT INTO genebank (name) VALUES ('Gotlandskanin');
+
+	-- Stub herd data
+	INSERT INTO herd (genebank_id, herd)
+	SELECT	DISTINCT gb.genebank_id, d."Genb"
+	FROM	genebank gb
+	JOIN	data d ON (TRUE)
+	WHERE	gb.name = 'Gotlandskanin'
+	ORDER BY d."Genb";
+
+	-- Stub individual data
+	INSERT INTO individual (origin_herd_id,
+		name, certificate, number, sex, birth_date,
+		colour_id, colour_note, death_note, litter, notes)
+	SELECT	h.herd_id,
+		d."Namn", d."Intyg", d."Nummer", d."Kön", d."Född",
+		d."Färgnr", d."Färg", d."Död", d."Kull", d."Övrigt"
+	FROM	genebank gb
+	JOIN	herd h ON (h.genebank_id = gb.genebank_id)
+	JOIN	data d ON (d."Genb" = h.herd)
+	WHERE	gb.name = 'Gotlandskanin'
+	ORDER BY d."Nummer";
+
+	-- Set mothers
+	-- Assumes that individual.number is unique
+	UPDATE individual i
+	SET	mother_id = parent.individual_id
+	FROM	individual parent
+	JOIN	data d ON (d."Mor nr" = parent.number)
+	WHERE	d."Nummer" = i.number;
+
+	-- Set fathers
+	-- Assumes that individual.number is unique
+	UPDATE individual i
+	SET	father_id = parent.individual_id
+	FROM	individual parent
+	JOIN	data d ON (d."Far nr" = parent.number)
+	WHERE	d."Nummer" = i.number;
+
+	-- Initial herd tracking
+	INSERT INTO herd_tracking (herd_id, individual_id, herd_tracking_date)
+	SELECT	i.origin_herd_id, i.individual_id, i.birth_date
+	FROM	genebank gb
+	JOIN	herd h ON (h.genebank_id = gb.genebank_id)
+	JOIN	individual i ON (i.origin_herd_id = h.herd_id)
+	WHERE	gb.name = 'Gotlandskanin'
+	ORDER BY i.individual_id;
+END_SQL
+
+# Load tracking data for years 2000 through to 2020
+year=2000
+while [ "$year" -le 2020 ]; do
+	cat <<-END_SQL
+		-- Fix column type etc.
+		ALTER TABLE data ALTER "$year" TYPE VARCHAR(10);
+		UPDATE data SET "$year" = CONCAT('G', "$year")
+		WHERE "$year" IS NOT NULL AND "$year" NOT LIKE 'G%';
+
+		-- Add missing herds
+		INSERT INTO herd (genebank_id, herd)
+		SELECT	DISTINCT gb.genebank_id, d."$year"
+		FROM	genebank gb
+		JOIN	data d ON (TRUE)
+		WHERE	gb.name = 'Gotlandskanin'
+		AND	d."$year" NOT IN (
+			SELECT herd
+			FROM	herd
+		)
+		ORDER BY d."$year";
+
+		-- Load $year data
+		INSERT INTO herd_tracking (herd_id, individual_id, herd_tracking_date)
+		SELECT	h.herd_id, i.individual_id, '$year-12-31'
+		FROM	genebank gb
+		JOIN	herd h ON (h.genebank_id = gb.genebank_id)
+		JOIN	data d ON (d."$year" = h.herd)
+		JOIN	individual i ON (i.number = d."Nummer")
+		WHERE	gb.name = 'Gotlandskanin'
+		ORDER BY i.individual_id;
+	END_SQL
+
+	year=$(( year + 1 ))
+done | psql
+
+# The Gotland data set has herd names etc. in a separate CSV file.
+# Load that file separately (using static filename here for now).
+
+csvsql	--db "postgresql://$PGUSER:dummy@/$PGDATABASE" \
+	--tables data2 \
+	--overwrite \
+	--skip-lines 5 \
+	--insert herd-registry-gotland.csv
```

Comment:
There is no information in the Readme about this file

Reply:
Will fix that later in the afternoon!

Reply:
This is now addressed in `README`, I hope.

Reply:
Using in2csv --skip-lines 5 worked for that file xlsx file I think! Should we specify the name on the cmd line also?

Reply:
This (using `--skip-lines 5`) is what I did in the update version.

Since the auxiliary file is not expected to change very often, I opted for hard-coding its name in the script.  This is in contrast to the other data files. These may change often.

Also, I kinda hope these loading script should be of _temorary_ use until such time that data entry happens via the web site.